lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -259,6 +260,39 @@ defmodule ExUnit.Runner do
     end
   end
 
+  defp maybe_create_tmp_dir(test, context, %{tmp_dir: true}) do
+    create_tmp_dir!(test, "", context)
+  end
+
+  defp maybe_create_tmp_dir(test, context, %{tmp_dir: path}) when is_binary(path) do
+    create_tmp_dir!(test, path, context)
+  end
+
+  defp maybe_create_tmp_dir(_, _, %{tmp_dir: other}) do
+    raise ArgumentError, "expected :tmp_dir to be atom true or a string, got: #{inspect(other)}"
+  end
+
+  defp maybe_create_tmp_dir(_, context, _) do
+    context
+  end
+
+  defp create_tmp_dir!(test, extra_path, context) do
+    module = escape_path(inspect(test.module))
+    name = escape_path(to_string(test.name))
+    path = Path.join(["tmp", module, name, extra_path])
```

Comment:
Should we include `tmp/` in the .gitignore of new files generated by `mix new`?

Reply:
Good call